Mathlib/Probability/Kernel/Disintegration/Basic.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Kin Yau James Wong. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Kin Yau James Wong, Rémy Degenne
+-/
+import Mathlib.Probability.Kernel.MeasureCompProd
+
+/-!
+# Disintegration of measures and kernels
+
+This file defines predicates for a kernel to "disintegrate" a measure or a kernel. This kernel is
+also called the "conditional kernel" of the measure or kernel.
+
+A measure `ρ : Measure (α × Ω)` is disintegrated by a kernel `ρCond : Kernel α Ω` if
+`ρ.fst ⊗ₘ ρCond = ρ`.
+
+A kernel `ρ : Kernel α (β × Ω)` is disintegrated by a kernel `κCond : Kernel (α × β) Ω` if
+`κ.fst ⊗ₖ κCond = κ`.
+
+## Main definitions
+
+* `MeasureTheory.Measure.IsCondKernel ρ ρCond`: Predicate for the kernel `ρCond` to disintegrate the
+  measure `ρ`.
+* `ProbabilityTheory.Kernel.IsCondKernel κ κCond`: Predicate for the kernel `κ Cond` to disintegrate
+  the kernel `κ`.
+
+Further, if `κ` is an s-finite kernel from a countable `α` such that each measure `κ a` is
+disintegrated by some kernel, then `κ` itself is disintegrated by a kernel, namely
+`ProbabilityTheory.Kernel.condKernelCountable`.
+
+## See also
+
+`Mathlib.Probability.Kernel.Disintegration.StandardBorel` for a **construction** of disintegrating
+kernels.
+-/
+
+open MeasureTheory Set Filter MeasurableSpace ProbabilityTheory
+open scoped ENNReal MeasureTheory Topology
+
+variable {α β Ω : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β} [MeasurableSpace Ω]
```

Comment:
It looks like in that file the measurable space on Omega can be inferred from the measures or kernels, so implicit is better:
```suggestion
variable {α β Ω : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β} {mΩ : MeasurableSpace Ω}
```
If that's not always the case, please add a typeclass argument in the lemmas where it is needed.